File: GoldenTicket/GoldenTicket/Entities/Message.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using GoldenTicket.Entities;

namespace GoldenTicket.Entities
{
    [Table("tblMessage")]
    public class Message
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int MessageID { get; set; }
        [Required]
        public int ChatroomID { get; set; }

        [ForeignKey("ChatroomID")]
        public Chatroom? Chatroom;

        [Required]
        public int SenderID { get; set; }

        [ForeignKey("SenderID")]
        public User? Sender { get; set; }

        [Required]
        public string? MessageContent { get;set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: golden_ticket_enterprise/lib/entities/Message.dart

import 'package:golden_ticket_enterprise/entities/user.dart';

class Message {
  final int messageID;
  final String messageContent;
  final User sender;
  final DateTime createdAt;
  Message({required this.messageID, required this.sender, required this.messageContent, required this.createdAt});

  factory Message.fromJson(Map<String, dynamic> json) {
    dynamic userData = json;
    return Message(
        messageID: json['messageID'],
        messageContent: json['messageContent'],
        sender: User.fromJson(json['sender']),
        createdAt: DateTime.parse(json['createdAt'])
    );
  }

}